pytest_django/fixtures.py

```diff
@@ -354,10 +354,13 @@ def finalize(self):
 
 
 @pytest.yield_fixture()
-def settings():
+def settings(request):
     """A Django settings object which restores changes after the testrun"""
     skip_if_no_django()
 
+    if 'transactional_db' in request.fixturenames:
```

Comment:
I suggest adding a comment explaining the rationale:

```suggestion
    # Order the `settings` fixture after DB.
    # We don't want overridden settings to be in effect during
    # DB setup/teardown/post_migrate.
    if 'transactional_db' in request.fixturenames:
```